pkg/packages/imports.go

```diff
@@ -38,20 +38,29 @@ type Imports struct {
 // UseType adds the package of given type to the import map and returns the alias
 // you can use in that Go file.
 func (m *Imports) UseType(in string) string {
+	if strings.HasPrefix(in, "map") {
```

Comment:
I like this idea. Maybe have a `switch` here for `[]` prefix as well, which would simplify the logic down in the stack?